src/runtime_src/core/pcie/windows/device_windows.cpp

```diff
@@ -520,7 +521,7 @@ struct bdf
     }
 
     auto& bdf = (*it).second;
-    return std::make_tuple(bdf.bus,bdf.device,bdf.function);
+    return std::make_tuple(bdf.domain, bdf.bus, bdf.device, bdf.function);
```

Comment:
Either convert to a struct under query::pcie_bdf or leave as tuple as done here, but do not mix tuple and the enum.

Reply:
In this line, there is no enum used, the variable "bdf" is array of uint16 received from windows/mgmt.cpp.
Using this array, a tuple is made.

Reply:
I understand, my point is that the enum is useless and doesn't provide any safety at all.  Either convert to embedded struct pcie_bdf { struct dbdf { ... }; ... }; 
or stay with the tuple as before adding the enum.   At least get rid of the enum under all circumstances.

Reply:
removed enum and stay wit tuple for now. Please clean up this part in next PRs.